packages/data-stores/src/queries/use-site-domains-query.ts

```diff
@@ -0,0 +1,109 @@
+import { UseQueryOptions, useQuery } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+export interface DomainData {
+	primary_domain: boolean;
+	blog_id: number;
+	subscription_id: string;
+	can_manage_dns_records: boolean;
+	can_manage_name_servers: boolean;
+	can_update_contact_info: boolean;
+	cannot_manage_dns_records_reason: unknown;
```

Comment:
Note that fields unused fields which I haven't dug into yet have just been left as `unknown`. We can look into what their type is once we actually need to use that field.